### DISCUSSION TITLE
How can the `globalID` arg accept an interface type?

### DISCUSSION THREAD

**Discussion by LegendRoland-K** [hayes/pothos#1176] [Q&A]
I have a `prismaInterface` called `Animal` then a variant called `Lion`. However, when I'm creating an `inputType` how can I restrict the `globalID` arguments for all types extending the `Animal` interface? 

```
export const Animal = builder.prismaInterface("Animal", {
  name: "Animal",
  fields: (t) => ({
    name: t.exposeString("name"),
  }),
  resolveType: (animal) => animal.type,
})


export const Lion = builder.prismaNode("Animal", {
  id: { field: "id" },
  variant: "Lion",
  interfaces: [Animal],
  fields: (t) => ({
    kills: t.exposeInt("kills"),
  }),
})

// Note, this does not work as it complains about the interface
const SomeInput = builder.inputType("SomeInput", {
  fields: (t) => ({
    animalId: t.globalID({ required: true , for: Animal}),
    fooBar: t.string({ required: false }),
  }),
})
```

I get the following error `Types of property 'kind' are incompatible. Type '"Interface"' is not assignable to type '"Object"'.ts(2322)`

I know I can define a list of types and pass it to `for`, however, would like if I would not have to hard-code all the types.

**Comment by hayes** [comment]
This isn't currently something that's supported, but you can just entirely omit the for option
